File: algorithms/581. Shortest Unsorted Continuous Subarray/solution.cc
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        int start = 1, end = 0, mini = INT_MAX, maxi = INT_MIN;
        for ( int i = 0; i < nums.size(); ++i ) {
            maxi = max(nums[i], maxi);
            if ( nums[i] < maxi ) {
                end = i;
            }
        }
        for ( int i = nums.size() - 1; i >= 0; --i ) {
            mini = min(nums[i], mini);
            if ( nums[i] > mini ) {
                start = i;
            }
        }
        return end - start + 1;
    }
};

File: algorithms/581. Shortest Unsorted Continuous Subarray/solution.py
class Solution(object):
    def findUnsortedSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        start, end, mini, maxi = 1, 0, float('inf'), float('-inf')
        for i in xrange(len(nums)):
            maxi = max(maxi, nums[i])
            if nums[i] < maxi:
                end = i
        for i in xrange(len(nums) - 1, -1, -1):
            mini = min(mini, nums[i])
            if nums[i] > mini:
                start = i
        return end - start + 1
